nixos/modules/services/misc/sourcehut/service.nix

```diff
@@ -0,0 +1,66 @@
+{ config, pkgs, lib }:
+serviceCfg: serviceDrv: iniKey: attrs:
+let
+  cfg = config.services.sourcehut;
+  cfgIni = cfg.settings."${iniKey}";
+  pgSuperUser = config.services.postgresql.superUser;
+
+  setupDB = pkgs.writeScript "${serviceDrv.pname}-gen-db" ''
+    #! ${cfg.python}/bin/python
+    from ${serviceDrv.pname}.app import db
+    db.create()
+  '';
+in
+with serviceCfg; with lib; recursiveUpdate
+{
+  environment.HOME = statePath;
+  path = [ config.services.postgresql.package ] ++ (attrs.path or [ ]);
+  restartTriggers = [ config.environment.etc."sr.ht/config.ini".source ];
+  serviceConfig = {
+    Type = "simple";
+    User = user;
+    Group = user;
+    Restart = "always";
+    WorkingDirectory = statePath;
+  } // (if (cfg.statePath == "/var/lib/sourcehut/${serviceDrv.pname}") then {
+          StateDirectory = [ "sourcehut/${serviceDrv.pname}" ];
+        } else {})
+  ;
```

Comment:
```suggestion
  } // (lib.optionalAttrs (cfg.statePath == "/var/lib/sourcehut/${serviceDrv.pname}") {
    StateDirectory = [ "sourcehut/${serviceDrv.pname}" ];
  };
```

Reply:
Additionally, I think that `mkMerge` would be slightly easier to read here:
```
  serviceConfig = mkMerge [
    {
      Type = "simple";
      User = user;
      Group = user;
      Restart = "always";
      WorkingDirectory = statePath;
    }
    (mkIf (cfg.statePath == "/var/lib/sourcehut/${serviceDrv.pname}") {
      StateDirectory = [ "sourcehut/${serviceDrv.pname}" ];
    })
  ];
```

Reply:
I tried it that way, but then subsequent uses of that serviceConfig didn't seem to work right. Got systemd complaints about missing an ExecStart, so my guess is a merge was overwriting something.

Reply:
Don't use `//` on `serviceConfig`, go for `mkMerge` like my example shows. That should solve your problem.

Reply:
Still don't seem to be able to make this work.